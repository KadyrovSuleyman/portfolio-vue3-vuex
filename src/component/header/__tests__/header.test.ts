import { mount, VueWrapper } from '@vue/test-utils';
import { createStore } from 'vuex';
import Header from '../header.vue';

let wrapper: VueWrapper<any>;
afterEach(() => {
  wrapper.unmount();
});

it('navButton renders', () => {
  wrapper = mount(Header, { global: { plugins: [createStore({})] } });
  expect(wrapper.element.outerHTML).toMatchSnapshot();
});

it('watchs props changes', async () => {
  wrapper = mount(Header, { global: { plugins: [createStore({})] } });
  expect(wrapper.find('header').classes()).toEqual(['header']);

  await wrapper.setProps({
    mods: {
      selected: true,
    },
  });
  expect(wrapper.find('header').classes()).toEqual(['header', 'header__selected']);

  await wrapper.setProps({
    mods: {
      selected: false,
    },
  });
  expect(wrapper.find('header').classes()).toEqual(['header']);
});
