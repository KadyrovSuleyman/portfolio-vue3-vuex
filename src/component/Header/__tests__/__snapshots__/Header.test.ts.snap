// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`navButton renders 1`] = `
<header class="header">
  <div class="header-navPanel">
    <div class="navPanel-navButton navPanel-navButton__selected"><a href="/stacking" class="navButton-link navButton-link__selected">Stacking</a>
      <div class="navButton-rectangle navButton-rectangle__selected"></div>
    </div>
    <div class="navPanel-navButton"><a href="/bridge" class="navButton-link">Bridge</a>
      <div class="navButton-rectangle"></div>
    </div>
    <div class="navPanel-navButton"><a href="/sho" class="navButton-link">SHO</a>
      <div class="navButton-rectangle"></div>
    </div>
  </div>
  <div class="header-walletBlock">
    <div class="walletBlock-connectBlock"><button class="connectBlock-button"> Connect wallet </button></div>
  </div>
</header>
`;
